typescript-sdk/src/evaluation/__tests__/run-evaluation.test.ts

```diff
@@ -0,0 +1,170 @@
+      json: async () => ({ ...baseSkipped }),
+    });
+    const result = await runEvaluation(details as any);
+    expect(result.status).toBe('skipped');
+    expect(result.details).toBe('skipped');
+  });
+
+  it('returns error result', async () => {
+    mockFetch.mockResolvedValueOnce({
+      ok: true,
+      json: async () => ({ ...baseError }),
+    });
+    const result = await runEvaluation(details as any);
+    expect(result.status).toBe('error');
+    if (result.status === 'error') {
+      expect(result.details).toBe('fail');
+      expect(result.error_type).toBe('EvalError');
+      expect(result.traceback).toEqual(['trace']);
+    } else {
+      throw new Error('Expected error result');
+    }
+  });
+
+  it('returns unknown status as error', async () => {
+    mockFetch.mockResolvedValueOnce({
+      ok: true,
+      json: async () => ({ status: 'weird' }),
+    });
+    const result = await runEvaluation(details as any);
+    expect(result.status).toBe('error');
+    if (result.status === 'error') {
+      expect(result.error_type).toBe('UnknownStatus');
+      expect(result.details).toContain('Unknown evaluation status');
+    } else {
+      throw new Error('Expected error result');
+    }
+  });
+
+  it('throws LangWatchApiError on non-ok response', async () => {
+    mockFetch.mockResolvedValueOnce({ ok: false, json: async () => ({}), status: 400, statusText: 'Bad', headers: { get: () => 'application/json' } });
+    await expect(runEvaluation(details as any)).rejects.toBeInstanceOf(LangWatchApiError);
+  });
+
+  it('propagates fetch errors', async () => {
+    mockFetch.mockRejectedValueOnce(new Error('network fail'));
+    await expect(runEvaluation(details as any)).rejects.toThrow('network fail');
+  });
+
+  it('calls setInput if canAutomaticallyCaptureInput is true', async () => {
+    vi.resetModules();
+    vi.doMock('../../client', () => ({
+      canAutomaticallyCaptureInput: () => true,
+      getApiKey: () => 'test-key',
+      getEndpoint: () => 'https://api',
+    }));
+    const span = {
+      setType: vi.fn(),
+      setInput: vi.fn(),
+      setMetrics: vi.fn(),
+      setOutputEvaluation: vi.fn(),
+      recordException: vi.fn(),
+      end: vi.fn(),
+      spanContext: () => ({ traceId: 'trace', spanId: 'span' }),
+    };
+    mockStartActiveSpan.mockImplementationOnce((name, fn) => fn(span));
+    mockFetch.mockResolvedValueOnce({ ok: true, json: async () => ({ ...baseProcessed }) });
+    const { runEvaluation: runEval } = await import('../run-evaluation');
+    await runEval(details as any);
+    expect(span.setInput).toHaveBeenCalledWith(expect.objectContaining({ trace_id: 'trace' }));
+  });
+
+  it('does not call setInput if canAutomaticallyCaptureInput is false', async () => {
+    vi.resetModules();
+    vi.doMock('../../client', () => ({
+      canAutomaticallyCaptureInput: () => false,
+      getApiKey: () => 'test-key',
+      getEndpoint: () => 'https://api',
+    }));
+    const span = {
```

Comment:
This seems a bit redundant. Maybe consider something like https://github.com/thoughtbot/fishery

Reply:
I hate writing tests, but this looks great. i'll have a look, thanks!!

Reply:
but happy to be shown i'm wrong